sigstore/_cli.py

```diff
@@ -82,6 +82,37 @@ def _boolify_env(envvar: str) -> bool:
         raise ValueError(f"can't coerce '{val}' to a boolean")
 
 
+def _set_default_verify_subparser(parser: argparse.ArgumentParser, name: str) -> None:
+    """
+    An argparse patch for configuring a default subparser for `sigstore verify`.
+
+    Adapted from <https://stackoverflow.com/a/26379693>
+    """
+    subparser_found = False
+    for arg in sys.argv[1:]:
+        if arg in ["-h", "--help"]:  # global help if no subparser
+            break
+    else:
+        for x in parser._subparsers._actions:  # type: ignore[union-attr]
+            if not isinstance(x, argparse._SubParsersAction):
+                continue
+            for sp_name in x._name_parser_map.keys():
+                if sp_name in sys.argv[1:]:
+                    subparser_found = True
+        if not subparser_found:
+            try:
+                # If `sigstore verify identity` wasn't passed explicitly, we need
+                # to insert the `identity` subcommand into the correct position
+                # within `sys.argv`. To do that, we get the index of the `verify`
+                # subcommand, and insert it directly after it.
+                verify_idx = sys.argv.index("verify")
+                sys.argv.insert(verify_idx + 1, name)
```

Comment:
Can we maybe log a warning here? I realise we don't want to make a breaking change but perhaps we can at least mark it was deprecated and slate it for future removal.

Reply:
Yeah, a log sounds good. I'll do that in a moment.

Reply:
https://github.com/sigstore/sigstore-python/pull/379/commits/088714b32e5d9d08a531d7c8b7ec73b7efd803ba